src/RemoteAPI/related/class-related-proxy.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Parsely Related REST API Proxy
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\RemoteAPI;
+
+use Parsely\Parsely;
+use WP_Error;
+
+/**
+ * Proxy for the /related endpoint.
+ *
+ * @since 3.2.0
+ */
+class Related_Proxy implements Proxy {
+	const RELATED_API_ENDPOINT = 'https://api.parsely.com/v2/related';
+
+	/**
+	 * Parsely Instance.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * The query arguments to send to the remote API.
+	 *
+	 * @var array
+	 */
+	private $query;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Parsely instance.
+	 * @param array   $query The query arguments to send to the remote API.
+	 * @since 3.2.0
+	 */
+	public function __construct( Parsely $parsely, array $query ) {
+		$this->parsely = $parsely;
+		$this->query   = $query;
+	}
+
+	/**
+	 * Accessor for private query var.
+	 *
+	 * @since 3.2.0
+	 */
+	public function get_query() {
+		return $this->query;
+	}
+
+	/**
+	 * Get URL for Recommendation API (GET /related).
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return string
+	 */
+	public function get_api_url(): string {
+		$query           = $this->query;
+		$query['apikey'] = $this->parsely->get_api_key();
+		$query           = array_filter( $query );
+
+		// Sort by key so the query args are in alphabetical order.
+		ksort( $query );
+
+		$query = apply_filters( 'wp_parsely_related_endpoint_args', $query );
+		return add_query_arg( $query, self::RELATED_API_ENDPOINT );
+	}
+
+	/**
+	 * Implements the fetcher for the proxy interface.
+	 *
+	 * @since 3.2.0
+	 */
+	public function get_items() {
+		$full_api_url = $this->get_api_url();
+
+		$result = wp_safe_remote_get( $full_api_url, array() );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$body    = wp_remote_retrieve_body( $result );
+		$decoded = json_decode( $body );
```

Comment:
All of the decoding, and checks, are technically a new task from the actual getting a response, so it could be handled with a `JSON_Decoder` class (which implements a `Decoder` interface, to allow for future expansion of retrieving XML, CSV, plain text, etc.). This would then centralise "the ability to decode some JSON and return an error if it fails" into one place, allowing multiple proxy classes to make use of it.

Reply:
Since we now have a single proxy point, then this line will only appear in one place - this could be switched out later on, but it's not a blocker to this PR.